### PR TITLE
Make tgmath tests generic over TGMath protocol, extend to Float80

### PR DESCRIPTION
**Pull request by stephentyrone**
Define most of the test machinery in terms of a internal TGMath protocol that refines BinaryFloatingPoint. Add a little bit of gyb support to automate adding additional types, as it is likely these tests will need to support Float16 and Float128 in the future. Make all the input values exactly representable in all of the types in question, to avoid representation errors, loosen default tolerance to 3 to make porting to new platforms less noisy.

### CODE REVIEW

**Comment by stephentyrone** [comment]
@swift-ci Please smoke test

**Comment by stephentyrone** [comment]
@swift-ci Please smoke test Linux.

**Comment by stephentyrone** [comment]
@swift-ci please smoke test.

File: test/stdlib/tgmath.swift.gyb [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+// REQUIRES: executable_test
+
+#if canImport(Darwin)
+  import Darwin.C.tgmath
+#else
+  import Glibc
+#endif
+
+#if (arch(i386) || arch(x86_64)) && !os(Windows)
+  typealias TestLiteralType = Float80
```
Comment (xwu) [inline test/stdlib/tgmath.swift.gyb]: Can this be simply `_MaxBuiltinFloatType`?

**Review by xwu — COMMENTED** [review]
LGTM

File: test/stdlib/tgmath.swift.gyb (hunk 1 above)
Comment (stephentyrone) [reply test/stdlib/tgmath.swift.gyb]: Probably.

**Comment by stephentyrone** [comment]
@swift-ci please smoke test and merge.

**Comment by stephentyrone** [comment]
@swift-ci please smoke test

File: test/stdlib/tgmath.swift.gyb (hunk 1 above)
Comment (stephentyrone) [reply test/stdlib/tgmath.swift.gyb]: Guess not =)

**Comment by stephentyrone** [comment]
@swift-ci please smoke test and merge.

**Comment by stephentyrone** [comment]
@swift-ci Please smoke test.

**Comment by stephentyrone** [comment]
@swift-ci Please smoke test.

**Comment by stephentyrone** [comment]
Sure enough, we should remove the ldexp test, since we've deprecated it and scalbn is the thing now.

**Comment by stephentyrone** [comment]
@swift-ci Please smoke test.

**Comment by tbkka** [comment]
This no longer checks that `CGFloat` implementations match `Float` on 32-bit and `Double` on 64-bit.  Was that intentional?

**Comment by stephentyrone** [comment]
That’s not what this test is for; ~everything will fail to build if that’s ever violated anyway, so it probably doesn’t need a test, but it’s fairly well covered already.

**Pull request merged**